scripts/operations_tests/operations_tests.py

```diff
@@ -88,193 +89,215 @@ def process_params(param):
-        elif test_case.status == "new baseline":
-            new_baselines += 1
-            if args.verbose:
-                print(f"[NEW] {test_case.operation} test #{test_case.test_number:03d}, {test_case.test_name}\n")
-
-    print(f"{failures} failed\n{passes} passed\n{new_baselines} new baseline(s)")
-    print(f"{'=' * 42}END{'=' * 42}")
-
-
-def time_mode(runs):
-    durations = defaultdict(list)
-    image_stack = load_image_stack()
-    for operation, test_case_info in TEST_CASES.items():
-        print(f"Running tests for {operation}:")
-        cases = test_case_info["cases"]
-        for case in cases:
-            sub_test_name = case["test_name"]
-            test_name = f"{operation.lower()}_{sub_test_name}"
-            params = case["params"] | test_case_info["params"]
-            op_func = FILTERS[operation].filter_func
-            for _ in range(runs):
-                image_stack2 = image_stack.copy()
-                duration = time_operation(image_stack2, op_func, params)[0]
-                durations[test_name].append(duration)
-
-    print_time_mode_results(durations)
-
-
-def print_time_mode_results(durations):
-    print(f"\n{'=' * 40}RESULTS{'=' * 40}", end="")
-    with open("timings.csv", "a", newline="", encoding="utf_8") as csvfile:
-        writer = csv.writer(csvfile, delimiter=",")
-        if csvfile.tell() == 0:
-            writer.writerow(["test_name", "avg", "quickest", "slowest", "stdev", "version", "commit_date"])
-        for test_name, times in durations.items():
-            avg = sum(times) / len(times)
-            quickest = min(times)
-            slowest = max(times)
-            print(f"\n{test_name}:\n\tAvg: {avg}\n\tQuickest: {quickest}\n\tSlowest: {slowest}")
-            data = (
-                test_name,
-                avg,
-                quickest,
-                slowest,
-                stdev(times),
-                GIT_TOKEN,
-                COMMIT_DATE,
-            )
-            writer.writerow(data)
-
-    print(f"{'=' * 42}END{'=' * 42}")
-
-
-def time_operation(image_stack, op_func, params):
-    start = time.perf_counter()
-    image_stack = run_operation(image_stack, op_func, params)
-    duration = time.perf_counter() - start
-    return duration, image_stack
-
-
-def run_test(test_case):
-    image_stack = load_image_stack()
-    test_case.duration, new_image_stack = time_operation(image_stack, test_case.op_func, test_case.params)
-    file_name = config_manager.save_dir / (test_case.test_name + ".npz")
-
-    if file_name.is_file():
-        baseline_image_stack = load_post_operation_image_stack(file_name)
-        compare_image_stacks(baseline_image_stack, new_image_stack.data, test_case)
-
-        if test_case.status == "pass":
-            print(".", end="")
-        elif test_case.status == "fail":
-            print("F", end="")
+class TestRunner:
+
+    def __init__(self, args):
+        self.args = args
```

Comment:
As `args` only gets `None` passed in, you could drop it as an argument. e.g.

```python
    def __init__(self):
        self.args = None
```

and then you don't need to create `args` or `parser` in `main()`